### DISCUSSION TITLE
rez-env <package> in .bashrc results in infinite loop

### DISCUSSION THREAD

**Discussion by daveoy** [AcademySoftwareFoundation/rez#1480] [Q&A] [answered]
Hello all,

Rez newbie here but I have a few packages up and running and I'm pretty impressed with what I've been able to do so far!

I have a simple bashrc that contains four lines:

'''
<pseudocode>
export REZ_CONFIG_FILE
export PATH=/path/to/rez:$PATH
source rez/complete/complete.sh
</pseudocode>
'''

If I add a "rez-env <working_package>" to my bashrc above, it results in an infinite loop. Hitting ctrl-c shows multiple ">" chars at the front of the line suggesting recursion through environments. 

Does this ring any bells to anyone?

**Comment by JeanChristopheMorinPerso** [comment, marked as answer]
Hi @daveoy ! `rez-env` creates shells. So your bashrc runs rez-env, which creates a shell, which sources your bashrc, which runs rez-env, etc.

You could try to use `/.bash_profile` instead of bashrc. I think this will fix your problem. You could also check to see if you are already in a rez environment by checking any of the Rez standard environnement variables and only running rez-env when required.

**Reply by daveoy** [reply]
```
if [[ -z $REZ_PRODUCTION_ENVIRONMENT_ROOT ]]; then
  # drop into our base PRODUCTION_ENVIRONMENT environment
  rez-env -q production_environment
fi
```
perfectly sets up the environment for me. thank you so much for this!
